torch/testing/_internal/common_methods_invocations.py

```diff
@@ -9373,12 +9373,10 @@ def __call__(self, opinfo, device, dtype, requires_grad, **kwargs):
         _foreach_inputs_kwargs = {k: kwargs.pop(k, v) for k, v in _foreach_inputs_default_kwargs.items()}
         _foreach_inputs_kwargs["requires_grad"] = requires_grad
 
-        from torch.testing._internal.common_dtype import get_all_dtypes
-
         for num_tensors, ord, out_dtype in product(
             num_input_tensors,
             (0, 1, 2, -1, -2, float('inf'), float('-inf')),
-            [None] + get_all_dtypes(),
+            (None,) + (torch.complex128,) if dtype in complex_types() else (torch.float64,),
```

Comment:
what coverage does this remove?

Reply:
integers, complex64, float32, float16, and bfloat16. 
integers wouldn't be supported anyway, i.e., test_foreach's test_parity checks the error messages are identical.
some lower prec's could cause no direct cast or something

Reply:
I'd rather still test more + just have expected skips vs silently testing less

Reply:
I found it not really straightforward to give skips/xfail to some of the sample inputs generated.
Would you happen to have some reference?

Reply:
https://github.com/pytorch/pytorch/blob/95b2766864a02cdb6bc49b97a7f7e859b7167e70/torch/testing/_internal/common_methods_invocations.py#L9481-L9507

something like this but for norm

Reply:
I'm sure my previous response is misleading. The dtypes with if-else here doesn’t eliminate any of the existing test cases as it’s aimed to test the new kwarg. Inappropriate dtype to the kwarg just leads to an error, and unfortunately for meta test, all dtypes (meaning parameterization of inputs) end up being decorated as xfail if the dtype kwarg parameterization covers all the possible ones

Reply:
Ah this parametrizes out_dtype, which is new.